src/rust/ide/src/controller/visualization.rs

```diff
@@ -0,0 +1,256 @@
+//! Visualization controller.
+//!
+//! Ths Visualization Controller is Responsible identifying all the available visualizations
+//! natively embedded in IDE and available within the project's `visualization` folder.
+
+use crate::prelude::*;
+
+use enso_protocol::language_server;
+use graph_editor::component::visualization::class;
+use graph_editor::component::visualization::JsSourceClass;
+use std::rc::Rc;
+
+
+
+// =============
+// === Error ===
+// =============
+
+/// Enumeration of errors used in `Visualization Controller`.
+#[derive(Debug,Fail)]
+#[allow(missing_docs)]
+pub enum VisualizationError {
+    #[fail(display = "Visualization \"{}\" not found.", identifier)]
+    NotFound {
+        identifier : VisualizationIdentifier
+    },
+    #[fail(display = "JavaScript visualization \"{}\" failed to be instantiated.", identifier)]
+    InstantiationError {
+        identifier : VisualizationIdentifier
+    }
+}
+
+
+
+// ===============================
+// === VisualizationIdentifier ===
+// ===============================
+
+/// This enum is used to identify a visualization either in the project folder or natively
+/// embedded in IDE.
+#[derive(Clone,Debug,Display,Eq,PartialEq)]
+#[allow(missing_docs)]
+pub enum VisualizationIdentifier {
+    Embedded(String),
+    File(language_server::Path)
+}
+
+
+
+// ==============================
+// === EmbeddedVisualizations ===
+// ==============================
+
+#[allow(missing_docs)]
+pub type EmbeddedVisualizationName = String;
+
+/// Embedded visualizations mapped from name to source code.
+#[derive(Shrinkwrap,Debug,Clone,Default)]
+#[shrinkwrap(mutable)]
+pub struct EmbeddedVisualizations {
+    #[allow(missing_docs)]
+    pub map:HashMap<EmbeddedVisualizationName,Rc<class::Handle>>
+}
+
+
+
+
+// ==============
+// === Handle ===
+// ==============
+
+const VISUALIZATION_FOLDER:&str = "visualization";
+
+#[derive(Debug)]
+struct VisualizationControllerData {
+    language_server_rpc     : Rc<language_server::Connection>,
+    embedded_visualizations : EmbeddedVisualizations
+}
+
+/// Visualization Controller is responsible for listing and loading all the available
+/// visualizations on the project and the native ones embedded on IDE.
+#[derive(Debug,Clone,CloneRef)]
+pub struct Handle {
+    data : Rc<RefCell<VisualizationControllerData>>
+}
```

Comment:
The Rc for language_server connection does not have to be under RefCell, because it's already internal mutable. Therefore I suggest to simplify the structs this way:
```suggestion
/// Visualization Controller is responsible for listing and loading all the available
/// visualizations on the project and the native ones embedded on IDE.
#[derive(Debug,Clone,CloneRef)]
pub struct Handle {
    language_server_rpc     : Rc<language_server::Connection>,
    embedded_visualizations : Rc<RefCell<EmbeddedVisualizations>>
}
```
Then you need borrow only when obtaining embedded visualizations.